realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -351,6 +414,56 @@ private void clearScheduledAccessTokenRefresh() {
         }
     }
 
+    // Wrapper class for handling the async operations of the underlying SyncSession calling `async_wait_for_download_completion`
+    private static class WaitForServerChangesWrapper {
+
+        private final CountDownLatch waitForChanges = new CountDownLatch(1);
+        private volatile boolean resultReceived = false;
+        private Long errorCode = null;
+        private String errorMessage;
+
+        /**
+         * Block until the wait either completes or is terminated for other reasons.
+         */
+        public void waitForServerChanges() {
+            if (!resultReceived) {
+                try {
+                    waitForChanges.await();
+                } catch (InterruptedException e) {
```

Comment:
I think the `waitForServerChanges()` should throws a checked `InterruptedException`instead and move the call this in `RealmCache` to make it works better with `getInstanceAsync()`